src/EA.Iws.Web/Areas/NotificationApplication/Controllers/ShareNotificationController.cs

```diff
@@ -0,0 +1,192 @@
+﻿namespace EA.Iws.Web.Areas.NotificationApplication.Controllers
+{
+    using Core.Notification;
+    using Infrastructure;
+    using Infrastructure.Authorization;
+    using Prsd.Core.Mediator;
+    using Requests.Notification;
+    using Requests.SharedUsers;
+    using Requests.Users;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using System.Web.Mvc;
+    using ViewModels.ShareNotification;
+
+    [AuthorizeActivity(typeof(AddSharedUser))]
+    public class ShareNotificationController : Controller
+    {
+        private readonly IMediator mediator;
+
+        public ShareNotificationController(IMediator mediator)
+        {
+            this.mediator = mediator;
+        }
+
+        // If the previous controller is the notifications options page, then remove the temp data for shared owners
+        private void CheckSharedDataIsValid()
+        {
+            try
+            {
+                var previousController = (Request.UrlReferrer.Segments.Skip(3).Take(1).SingleOrDefault() ?? "Home").Trim('/');
+                if (previousController == "options")
+                {
+                    TempData.Remove("SharedUsers");
+                }
+            }
+            catch (Exception)
```

Comment:
Empty catch block. This should be handled appropriately or removed if not relevant.